### ISSUE TITLE
[Task]: Sort workspaces on homepage in fixed alphabetical order

### ISSUE THREAD

**Issue by pranayagarwal96** [appsmithorg/appsmith#31108]
### SubTasks

Currently, workspaces on homepage are sorted in order of last edited. However, basis feedback received from users, they would rather prefer to have the workspaces in fixed order, so that they don't have to keep searching in the list every time (as the order keeps dynamically changing basis usage)

**Comment by abhvsn** [comment]
Need to explicitly maintain the context when switching back to homepage, if we go ahead with this implementation https://theappsmith.slack.com/archives/C02K2MZERSL/p1709111336331609

**Comment by abhvsn** [comment]
Moving this to product backlog as we don't have the bandwidth to address the issue described above around maintaining the context, and this is not going to be a backend only code change.

**Comment by bookcaseballpen** [comment]
I think this sorting feature is critial.
our company ops currenlty also using appsmith admin tools. they are not tech savy. and they keep complainning the change of app locations.

**Comment by bookcaseballpen** [comment]
@pranayagarwal96 
Hi, 
I am looking for a  fix for both. 
but the applicaiton order inside a workspace is more critical than than the workspace order to our team.

**Comment by pranayagarwal96** [comment]
Got it - unfortunately, this is not in our near-term pipeline but we do intend to provide different sorting options to user for both apps and workspaces over time

**Comment by holylander** [comment]
I'd suggest to just have them sorted alphabetically ( very easy to implement in JS), plus also adding a small simple filter box, that search by workspace name

**Comment by daaliachhak17** [comment]
https://zemoso.atlassian.net/browse/TNOSB-431- ticket created in Jira

**Comment by MajaharZemoso** [comment]
@pranayagarwal96 
I am picking up this issue and i will be focusing on sorting workspace in alphabetical order.

**Comment by MajaharZemoso** [comment]
@Nikhil-Nandagopal @abhvsn @pranayagarwal96 
I wanted to clarify our approach: Are we planning to sort the applications also, or are we focusing solely on sorting by workspace?

**Comment by holylander** [comment]
my two cents is that having an fixed sorting helps UX. Sorting by default name is a good example, this way, on the next visit, user knows where the workspaces are etc.
On the other hand, a search box by name would be also very convenient

**Comment by Nikhil-Nandagopal** [comment]
@MajaharZemoso we can do it for apps next, let's start with a PR for workspaces

**Comment by MarkoSulamagi** [comment]
+1 

Looking forward to the solution.

- Ideally, user could drag/drop the order themselves (or can be configured by administrator, if that's simpler)
- As a quickfix, alphabetic or based on creation date ordering would be ok (better than today)

Thanks!

**Comment by jbaanus** [comment]
+1 from my side as well :)

**Comment by infinitetrooper** [comment]
Hey @MarkoSulamagi  - Just to clarify, are you looking for fixed order of applications in a workspace, or fixed order of workspaces in the sidebar or both?

**Comment by Nikhil-Nandagopal** [comment]
@MajaharZemoso are you still working on this?

**Comment by Naveen-Goud** [comment]
Hii @Nikhil-Nandagopal @pranayagarwal96 , can I pick this ticket ?

thank you

**Comment by riteshkew** [comment]
Hi @Naveen-Goud 
We have assigned this to you. Please go ahead with the required changes. Looking forward to your PR.

**Comment by riteshkew** [comment]
Hey @Naveen-Goud 
This is a reminder to add a reference to this ticket number in the PR. This would enable us to track it better internally.

**Comment by MajaharZemoso** [comment]
HI @Nikhil-Nandagopal 
Earlier, I was focused on another task, but I've now completed the subtask(sorting applications in workspace) and submitted the PR. Whenever you have a moment Please review it. Thanks in advance!

**Comment by AnweshPeddineni** [comment]
Hi @Nikhil-Nandagopal, @pranayagarwal96,  

I came across this issue and saw that it's labeled **"Inviting Contribution"**. I'm a new contributor and excited to start contributing to Appsmith! 

Could you please confirm if this issue is still open and if I can work on it? If there are any prerequisites or additional details I should know before starting, please let me know.  

Also, if you have any **open backend/server-side issues** that need contributions, I’d love to help! Please let me know if there’s anything suitable for me to work on.  

Looking forward to your guidance. Thanks! 😊

**Comment by tomjose92** [comment]
I would like to pick up this github issue, if its not being worked upon
